modules/srv6/datapath_local.c

```diff
@@ -0,0 +1,409 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2025 Olivier Gournet
+
+#include <gr_datapath.h>
+#include <gr_eth.h>
+#include <gr_fib6.h>
+#include <gr_graph.h>
+#include <gr_ip6_control.h>
+#include <gr_ip6_datapath.h>
+#include <gr_log.h>
+#include <gr_mbuf.h>
+#include <gr_srv6.h>
+#include <gr_trace.h>
+
+#include <rte_byteorder.h>
+#include <rte_ether.h>
+#include <rte_graph_worker.h>
+#include <rte_ip6.h>
+
+#include <netinet/in.h>
+
+//
+// references are to rfc8986
+//
+
+enum {
+	IP_INPUT = 0,
+	IP6_INPUT,
+	IP6_LOCAL,
+	INVALID_PACKET,
+	UNEXPECTED_UPPER,
+	NO_TRANSIT,
+	DEST_UNREACH,
+	EDGE_COUNT,
+};
+
+static const char *behavior_str[SR_BEHAVIOR_MAX] = {
+	[SR_BEHAVIOR_END] = "end",
+	[SR_BEHAVIOR_END_T] = "end.t",
+	[SR_BEHAVIOR_END_DT6] = "end.dt6",
+	[SR_BEHAVIOR_END_DT4] = "end.dt4",
+	[SR_BEHAVIOR_END_DT46] = "end.dt46",
+};
+
+static const char *behavior_to_str(gr_srv6_behavior_t b) {
+	return behavior_str[b];
+}
```

Comment:
probably not needed to have a function here. You can get the behavior name directly from the `behavior_str` array in `trace_srv6_format`.